### PR TITLE
ATS-995, MNT-23248: TIFF to PDF ImageMagick transform don't work

### PR DESCRIPTION
**Pull request by krdabrowski**
- introducing new custom transformer-misc: ImageToPdf

### CODE REVIEW

**Comment by montgolfiere** [comment]
I realise still Draft, but as per @krdabrowski 
> PR [...] related with TIFFs to PDFs conversion - it's temporarily a draft, but at the same time it's reviewable

QQ: Are you planning to add a test for multi=page TIFF to multi-page PDF ?

File: engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    }
+
+    private static void verifyOptions(final Integer startPage, final Integer endPage)
+    {
+        if (startPage != null && startPage < 0)
+        {
+            throw new IllegalArgumentException(NEGATIVE_START_PAGE_ERROR_MESSAGE);
+        }
+
+        if (endPage != null && endPage < 0)
```
Comment (tpage-alfresco) [inline engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java:208]: What does start or end page of zero mean? Is the first page number 0?

**Comment by krdabrowski** [comment]
TIFF file updated to 6 page file.

File: engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java (hunk 1 above)
Comment (krdabrowski) [reply engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java:208]: Yes, pages starts from 0.

File: engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    private void scaleAndDrawImage(final PDDocument pdfDocument, final BufferedImage bufferedImage, final String pdfFormat) throws IOException
+    {
+        final PDPage pdfPage = new PDPage(resolvePdfFormat(pdfFormat));
+        pdfDocument.addPage(pdfPage);
+        final PDImageXObject image = LosslessFactory.createFromImage(pdfDocument, bufferedImage);
+        try (PDPageContentStream pdfPageContent = new PDPageContentStream(pdfDocument, pdfPage))
+        {
+            final PDRectangle pageSize = pdfPage.getMediaBox();
+            final float withRatio = pageSize.getWidth() / image.getWidth();
+            final float heightRatio = pageSize.getHeight() / image.getHeight();
```
Comment (tpage-alfresco) [inline engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java:140]: Do we need to handle edge cases around dimensions being zero?

File: engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+    private void scaleAndDrawImage(final PDDocument pdfDocument, final BufferedImage bufferedImage, final String pdfFormat) throws IOException
+    {
+        final PDPage pdfPage = new PDPage(resolvePdfFormat(pdfFormat));
+        pdfDocument.addPage(pdfPage);
+        final PDImageXObject image = LosslessFactory.createFromImage(pdfDocument, bufferedImage);
+        try (PDPageContentStream pdfPageContent = new PDPageContentStream(pdfDocument, pdfPage))
+        {
+            final PDRectangle pageSize = pdfPage.getMediaBox();
+            final float withRatio = pageSize.getWidth() / image.getWidth();
```
Comment (tpage-alfresco) [inline engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java:139]: Typo:
```suggestion
            final float widthRatio = pageSize.getWidth() / image.getWidth();
```

File: engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java (hunk 2 above)
Comment (krdabrowski) [reply engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java]: Thanks! Fixed.

File: engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java (hunk 3 above)
Comment (krdabrowski) [reply engines/misc/src/main/java/org/alfresco/transform/misc/transformers/ImageToPdfTransformer.java]: Thanks.

**Review by tpage-alfresco — APPROVED** [review]
LGTM

**Review by gevagel — APPROVED** [review]

**Review by mpichura — APPROVED** [review]
LGTM. Nice parametrized unit tests :)

**Pull request merged**